src/Events/Custom_Tables/V1/Tables/Events.php

```diff
@@ -58,14 +58,15 @@ protected function get_update_sql() {
 		$table_name = self::table_name(true);
 		$charset_collate = $wpdb->get_charset_collate();
 
+		// VARCHAR(19) to store YYYY-MM-DD HH:MM:SS values as strings and allow partial compare.
 		return "CREATE TABLE `{$table_name}` (
 			`event_id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT,
 			`post_id` BIGINT(20) UNSIGNED NOT NULL,
-			`start_date` DATETIME NOT NULL,
```

Comment:
Not totally against this change. BUT from what I understand of the original issue, wouldn't another fix be to remove the NOT NULL constraint and allow nulls? Hate to move a date to a non-date type if we can avoid it.

Reply:
I've ran a test setting the field to `DATETIME` only, it will still throw issues.
The `NO_ZERO_DATE` or `NO_ZERO_IN_DATE` `sql_mode` will condition not only insertions, but selections as well. Just searching a `DATETIME` field under that condition as a string will trigger a warning.